pcsx2/PINE.cpp

```diff
@@ -121,23 +118,29 @@ PINEServer::PINEServer(SysCoreThread* vm, unsigned int slot)
 	// maximum queue of 4096 commands before refusing, approximated to the
 	// nearest legal value. We do not use SOMAXCONN as windows have this idea
 	// that a "reasonable" value is 5, which is not.
-	listen(m_sock, 4096);
+	if (listen(m_sock, 4096))
+	{
+		Console.WriteLn(Color_Red, "PINE: Cannot listen for connections! Shutting down...");
```

Comment:
This still isn't ideal, the rest of the application will still think PINE is active.. but oh well, better than it was I suppose.

You could move this to a boolean `Initialize()` method, or leave it until after merge (but that risks it never getting changed.. :-))

Reply:
Well all failures (`WSAStartup`, `socket`, `bind`) in the constructor just prints to log and returns. Perhaps the way to show failure to the rest of the application is to set `m_end` to true. The only parts of pcsx2 that know or care about PINE is the reload mechanism in `VMManager` so it's not like it's the end of the world. But I can try to make it more resilient to failures if you want.

Reply:
I have now made explicit `Initialize` and `Deinitialize`. The constructor is now empty and it's fine to have it constructed at all times so no more unique pointer reset and whatever to manage it. Not sure if more stuff should be moved to constructor but I kinda like it like this, made the buffer allocations could be moved there?